File: tutorials/iterative_closest_point/matlab/script_icp.m
% script_icp

%% read the data
px = csvread('../data/points_x.csv');
py = csvread('../data/points_y.csv');


objICP = IterativeClosestPoint(px,py);
objICP = objICP.register;

py_transformed = objICP.applyTransform(py,objICP.Transform);


%% plot points
% figure, plot3(px(:,1),px(:,2),px(:,3),'r.');
% hold on; plot3(py(:,1),py(:,2),py(:,3),'b.');
% hold on; plot3(py_transformed(:,1),py_transformed(:,2),py_transformed(:,3),'g.');
% axis equal

% plot surfaces
trix = csvread('../data/tris_x.csv');
triy = csvread('../data/tris_y.csv');

figure, axis equal, hold on;
h1 = patch('vertices',px,'faces',trix, ...
    'facecolor','r','edgecolor','r','edgealpha',.2,'facealpha',.2);
h2 = patch('vertices',py,'faces',triy, ...
    'facecolor','b','edgecolor','b','edgealpha',.2,'facealpha',.2);
h3 = patch('vertices',py_transformed,'faces',triy, ...
    'facecolor','g','edgecolor','g','edgealpha',.2,'facealpha',.2);
view(3); light; lighting phong;
legend([h1,h2,h3],'Fixed','Moving','Transformed')

File: tutorials/iterative_closest_point/python/script_icp.py
# script_icp

import numpy as np
from icp import IterativeClosestPoint as ICP
from matplotlib import pyplot as plt 
from mpl_toolkits.mplot3d import Axes3D


px = np.genfromtxt('../data/points_x.csv',delimiter=',')
py = np.genfromtxt('../data/points_y.csv',delimiter=',')

objICP = ICP(px,py)
objICP.register()
py_transformed = objICP.apply_transform(py,objICP.transform)


# plot
trix = np.genfromtxt('../data/tris_x.csv',delimiter=',')-1
triy = np.genfromtxt('../data/tris_y.csv',delimiter=',')-1

fig = plt.figure()
ax = fig.gca(projection='3d')
ax.plot_trisurf(px[:,0], px[:,1], px[:,2], triangles=trix, color='b')
ax.plot_trisurf(py[:,0], py[:,1], py[:,2], triangles=triy, color='r')
ax.plot_trisurf(py_transformed[:,0], py_transformed[:,1], py_transformed[:,2], triangles=triy, color='g')
plt.show()
# plt.savefig('icp.png')
